lib/index.js

```diff
@@ -3055,6 +3059,254 @@ UFDS.prototype.listResolvers = function listResolvers(region, cb, noCache) {
     this.search(dn, opts, cb, noCache);
 };
 
+/**
+ * AccessKeys related methods
+ */
+
+/**
+ * Adds a new AccessKey to a given user record.
+ *
+ * The AccessKeyId and AccessKeySecret will be generated.
+ *
+ * This method will return you the full key as processed by UFDS.
+ *
+ * @param {Object} user the user record you got from getUser.
+ * @param {String} (Optional) account uuid for a customer sub-user
+ * @param {Function} callback of the form fn(err, key).
+ * @throws {TypeError} on bad input.
```

Comment:
Any way to get that TypeError returned through the callback - I find it's bad form to mix throw/callback (note that I'm not talking about assert.X - as that should be a programmer error).

Same goes for following functions.

Reply:
Given this is a pattern repeated through the whole node-ufds program, sounds good if I modify this for every method there on a separate issue so it's clear what & why we're changing that?

Reply:
Certainly (I didn't know it already used that pattern).